### PR TITLE
vscode-extensions: add 2 extensions

### PR DESCRIPTION
**Pull request by rhoriguchi**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -3889,6 +3904,21 @@ let
...
 
+      zainchen.json = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "json";
+          publisher = "ZainChen";
+          version = "2.0.2";
+          sha256 = "sha256-nC3Q8KuCtn/jg1j/NaAxWGvnKe/ykrPm2PUjfsJz8aI=";
+        };
+        meta = {
+          description = "This extension adds json support for Visual Studio Code";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:3915]:
```suggestion
          changelog = "https://marketplace.visualstudio.com/items/ZainChen.json/changelog";
          description = "A Visual Studio Code extension for json support";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -718,6 +718,21 @@ let
...
 
+      cameron.vscode-pytest = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vscode-pytest";
+          publisher = "Cameron";
+          version = "0.1.1";
+          sha256 = "sha256-YU37a0Q+IXusXgwf9doxXLlYiyzkizbPjjdCZFxeDaA=";
+        };
+        meta = {
+          description = "A VS Code extensions that adds IntelliSense support for pytest fixtures";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:729]:
```suggestion
          description = "A Visual Studio Code extension that adds IntelliSense support for pytest fixtures";
```

**Comment by superherointj** [comment]
Please rebase.

**Comment by rhoriguchi** [comment]
> Please rebase.

Done

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -837,6 +837,22 @@ let
...
 
+      cameron.vscode-pytest = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vscode-pytest";
+          publisher = "Cameron";
+          version = "0.1.1";
+          sha256 = "sha256-YU37a0Q+IXusXgwf9doxXLlYiyzkizbPjjdCZFxeDaA=";
+        };
+        meta = {
+          description = "A Visual Studio Code extension that adds IntelliSense support for pytest fixtures";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:848]: Changelog is missing.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -837,6 +837,22 @@ let
...
+          changelog = "https://github.com/cameronmaske/pytest-vscode/blob/master/CHANGELOG.md";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:849]: `changelog` should come before `description`.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 5]
```diff
@@ -4640,6 +4656,22 @@ let
...
 
+      zainchen.json = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "json";
+          publisher = "ZainChen";
+          version = "2.0.2";
+          sha256 = "sha256-nC3Q8KuCtn/jg1j/NaAxWGvnKe/ykrPm2PUjfsJz8aI=";
+        };
+        meta = {
+          description = "This extension adds json support for Visual Studio Code";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:4667]:
```suggestion
          description = "A Visual Studio Code extension for json support";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 6]
```diff
@@ -4640,6 +4656,22 @@ let
...
+          changelog = "https://marketplace.visualstudio.com/items/ZainChen.json/changelog";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:4668]: `changelog` should come before `description`

**Review by superherointj — APPROVED** [review]

**Pull request merged**